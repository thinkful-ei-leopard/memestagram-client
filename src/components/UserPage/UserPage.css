.user-posts-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.user-posts {
  width: 200px;
  height: 200px;
  position: relative;
  overflow: hidden;
}

.user-meme {
  display: inline;
  margin: 0 auto;
  margin-left: -25%;
  height: 100%;
  width: auto;
}