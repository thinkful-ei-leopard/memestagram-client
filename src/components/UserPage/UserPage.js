import React from 'react'
import { Link } from 'react-router-dom'
import MemeContext from '../../context/MemeContext'
import PostsService from '../../services/posts-service'
import NavBar from '../NavBar/NavBar'

import './UserPage.css'

export default class UserPage extends React.Component {
  state = {
    posts: [],
    like: 'ü§ç'
  }

  static contextType = MemeContext

  componentDidMount(){
    const{ userId }=this.props.match.params
    this.context.clearError()
    PostsService.getUserPosts(userId)
      .then(res => this.setState({posts: res}))
      .catch(this.context.setError)
  }

  addLike() {
    if(this.state.like === 'ü§ç'){
     this.setState({
        like: '‚ù§Ô∏è',
        count: this.state.count + 1
     }) 
    } else if(this.state.like === '‚ù§Ô∏è'){
      this.setState({
        like: 'ü§ç',
        count: this.state.count - 1
      })   
    }
  }

  renderUserInfo() {
    const posts = this.state.posts
    
    if(!Array.isArray(posts) || !posts.length){
      return <p>Loading...</p>
    } else {
      return (
        <div>
          <div className='image-cropper'>
            <img src={posts[0].userImg} alt='user' className='user-img'></img>
          </div>
            <p className='username'>{posts[0].username}</p>
        </div>
        )
    }
  }
  
  renderPosts() {
    const posts = this.state.posts

    const postsArr = []
    if(!Array.isArray(posts) || !posts.length) {
      postsArr.push(
        <div className='no-posts'>
          <h2 className='no-posts-message'>No memes yet!</h2>
        </div>
        )
    } else {
      for(let i = 0; i < posts.length; i++) {
        postsArr.push(
          <div className='user-posts'>  
            <Link to={`/posts/${posts[i].id}`}>
              <img src={posts[i].memeImg} alt='meme' className='user-meme'></img>
            </Link>
          </div>
            )
        }
    }

    return postsArr
  }

render() {
  return (
    <div>
      <NavBar/>
      {this.renderUserInfo()}
      <div className='user-posts-container'>
        {this.renderPosts()}
      </div>
    </div>
  )
}
}