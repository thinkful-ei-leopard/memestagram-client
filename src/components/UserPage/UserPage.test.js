import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import UserPage from './UserPage';

describe('User Page Component', () => {
  
  const defaultProps = {
    match: {params: {postId: 1}}
  }

  it('UserPage renders without crashing', () =>{
    const div = document.createElement('div');
  ReactDom.render(
    <BrowserRouter>
      <UserPage {...defaultProps}/>
    </BrowserRouter>,
    div
  );
  ReactDom.unmountComponentAtNode(div);
  });

  it('renders User Page', () => {

    const wrapper = shallow(
      <BrowserRouter>
        <UserPage {...defaultProps}/>
      </BrowserRouter>
    )
    
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})