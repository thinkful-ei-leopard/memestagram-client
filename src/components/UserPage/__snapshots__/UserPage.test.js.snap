// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Page Component renders User Page 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <UserPage
    match={
      Object {
        "params": Object {
          "postId": 1,
        },
      }
    }
  />
</Router>
`;
